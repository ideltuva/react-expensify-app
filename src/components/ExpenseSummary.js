import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div>
        <p> -- </p>
        <p> Expense count: { props.expenses.length }  </p>
        <p> Total sum: { numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00') }  </p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);