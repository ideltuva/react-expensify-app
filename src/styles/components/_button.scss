.button {
    background: $light-blue;
    color: $snow-white;
    border: none;
    font-size: $font-size-large;
    display: inline-block;
    font-weight: 300;
    line-height: 1;
    padding: $s-size $m-size;
    text-decoration: none;
}

.button__link {
    background: none;
}

.button--secondary {
    background: #888888
}
