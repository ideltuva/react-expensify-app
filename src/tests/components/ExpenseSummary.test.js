import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test(`should render expense count `, () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot()
});

test(`should render expense sum`, () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot()
});



