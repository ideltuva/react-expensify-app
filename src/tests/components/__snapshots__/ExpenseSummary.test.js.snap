// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense count  1`] = `
<div>
  <p>
     -- 
  </p>
  <p>
     Expense count: 
    3
      
  </p>
  <p>
     Total sum: 
    $1,141.98
      
  </p>
</div>
`;

exports[`should render expense sum 1`] = `
<div>
  <p>
     -- 
  </p>
  <p>
     Expense count: 
    3
      
  </p>
  <p>
     Total sum: 
    $1,141.98
      
  </p>
</div>
`;
